torch/jit/annotations.py

```diff
@@ -63,6 +63,8 @@ def get_num_params(fn):
     if source is None:
         return None
     py_ast = ast.parse(source)
+    if len(py_ast.body) == 1 and isinstance(py_ast.body[0], ast.ClassDef):
+        raise RuntimeError("cannot create a module ({}) inside jit.script".format(py_ast.body[0].name))
```

Comment:
Well it's a class not a module, so this error message is really misleading. What's more, it mostly shows up when one tries to instantiate an object of a class from within a script function, so it's not very helpful.

Reply:
My bad, I thought since `Softmax` extends `Module`, that would be the appropriate message based on the example given in issue. Also, about the second point, what's the expected behavior?

Reply:
But I didn't check if it was an instance of `Module`, so yes, that's clearly an error. I'll make the change.